robot-agent/packageMonitor.js

```diff
@@ -0,0 +1,168 @@
+const { spawn } = require('child_process');
+const { EventEmitter } = require('events');
+
+const { getInstalledPackages } = require('./utils');
+
+const { getLogger } = require('@transitive-sdk/utils');
+const log = getLogger('packageMonitor.js');
+log.setLevel('info');
+
+class PackageMonitor extends EventEmitter {
+  constructor() {
+    super();
+    log.debug('PackageMonitor created');
+    this.installedPackages = {};
+    this.mqttSync = null;
+    this.AGENT_PREFIX = null;
+    this.initialized = false;
+    // for late subscribers raise events for already running packages
+    this.on('newListener', (event, listener) => {
+      log.debug(`New listener for event: ${event}`);
+      if (event === 'packageStarted') {
+        Object.entries(this.installedPackages).forEach(([pkgName, pkgData]) => {
```

Comment:
We usually use `lodash` for this:
`_.forEach( this.installedPackages, (pkgData, pkgName) => ...`

Also, this and the `packageStopped` case are virtually the same.